### ISSUE TITLE
JIRA Project name using reserved JIRA Jql word causes exception

### ISSUE THREAD

**Issue by cx-scord** [checkmarx-ltd/cx-flow#736]
### Description

When setting a JIRA Project key name in the config, if it uses a reserved a JIRA Jql word in the Jira project's key name, we will get an error.

I even tried using encoded characters directly in the YAML `jira.project`property config, like `%22` for quotes and `%4F%52` for the "OR" word, but it doesn't work.

### Expected Behavior

Issues should be created in JIRA in a project named after a Jql reserved word.

### Actual Behavior

Throws a `RestClientException`exception error when using a JIRA Jql reserved word from the list I found in a JIRA issue [here](https://jira.atlassian.com/browse/JRACLOUD-65837) and I found out it's hardcoded in the following JIRA Java API [JqlStringSupportImpl.java](https://docs.atlassian.com/software/jira/docs/api/5.2/com/atlassian/jira/jql/util/JqlStringSupportImpl.html#encodeStringValue(java.lang.String)) class. This class has methods that might help solve this issue by encoding each input with the an encoding method (I didn't test):

"abort", "access", "add", "after", "alias", "all", "alter", "and", "any", "as", "asc", "audit", "avg", "before", "begin", "between", "boolean", "break", "by", "byte", "catch", "cf", "char", "character", "check", "checkpoint", "collate", "collation", "column", "commit", "connect", "continue", "count", "create", "current", "date", "decimal", "declare", "decrement", "default", "defaults", "define", "delete", "delimiter", "desc", "difference", "distinct", "divide", "do", "double", "drop", "else", "empty", "encoding", "end", "equals", "escape", "exclusive", "exec", "execute", "exists", "explain", "false", "fetch", "file", "field", "first", "float", "for", "from", "function", "go", "goto", "grant", "greater", "group", "having", "identified", "if", "immediate", "in", "increment", "index", "initial", "inner", "inout", "input", "insert", "int", "integer", "intersect", "intersection", "into", "is", "isempty", "isnull", "join", "last", "left", "less", "like", "limit", "lock", "long", "max", "min", "minus", "mode", "modify", "modulo", "more", "multiply", "next", "noaudit", "not", "notin", "nowait", "null", "number", "object", "of", "on", "option", "or", "order", "outer", "output", "power", "previous", "prior", "privileges", "public", "raise", "raw", "remainder", "rename", "resource", "return", "returns", "revoke", "right", "row", "rowid", "rownum", "rows", "select", "session", "set", "share", "size", "sqrt", "start", "strict", "string", "subtract", "sum", "synonym", "table", "then", "to", "trans", "transaction", "trigger", "true", "uid", "union", "unique", "update", "user", "validate", "values", "view", "when", "whenever", "where", "while", "with"

### Reproduction

Using reserved words in the JIRA project key configuration will suffice to trigger it, like "OR" or "IN", for example:

```yaml
jira:
  - project: OR
```
```yaml
jira:
  - project: "OR"
```
**NOTE**: I didn't try with the `issue-type` field, but this might happen with any Jql input that has a reserved word included.

For example, if the name of the project key as `"OR"` or just `OR`, the exception stack is:

```
2021-05-14 13:13:35.042 ERROR 57660 --- [  scan-results1] c.checkmarx.flow.service.ResultsService  : Error occurred while processing results.
com.checkmarx.flow.exception.JiraClientRunTimeException: Wasn't able to publish the following issues into JIRA:
(...)
==============
due to the following reason: [ErrorCollection{status=400, errors={}, errorMessages=[Error in JQL Query: Expecting either a value, list or function but got 'OR'. You must surround 'OR' in quotation marks to use it as a value. (line 1, character 11)]}]
        at com.checkmarx.flow.service.ResultsService.throwExceptionWhenPublishingErrorOccurred(ResultsService.java:269)
        at com.checkmarx.flow.service.ResultsService.handleJiraRestClientException(ResultsService.java:256)
        at com.checkmarx.flow.service.ResultsService.handleJiraCase(ResultsService.java:231)
        at com.checkmarx.flow.service.ResultsService.processResults(ResultsService.java:137)
        at com.checkmarx.flow.service.ResultsService.publishCombinedResults(ResultsService.java:101)
        at com.checkmarx.flow.service.ResultsService$$FastClassBySpringCGLIB$$2328645f.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
        at org.springframework.aop.framework.adapter.MethodBeforeAdviceInterceptor.invoke(MethodBeforeAdviceInterceptor.java:56)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
        at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115)
        at org.springframework.aop.interceptor.AsyncExecutionAspectSupport.lambda$doSubmit$3(AsyncExecutionAspectSupport.java:276)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.atlassian.jira.rest.client.api.RestClientException: [ErrorCollection{status=400, errors={}, errorMessages=[Error in JQL Query: Expecting either a value, list or function but got 'OR'. You must surround 'OR' in quotation marks to use it as a value. (line 1, character 11)]}]
        at com.atlassian.jira.rest.client.internal.async.DelegatingPromise.claim(DelegatingPromise.java:45)
        at com.checkmarx.flow.service.JiraService.getIssues(JiraService.java:228)
        at com.checkmarx.flow.service.JiraService.process(JiraService.java:1211)
        at com.checkmarx.flow.service.ResultsService.handleJiraCase(ResultsService.java:229)
        ... 19 common frames omitted
Caused by: com.atlassian.jira.rest.client.api.RestClientException: [ErrorCollection{status=400, errors={}, errorMessages=[Error in JQL Query: Expecting either a value, list or function but got 'OR'. You must surround 'OR' in quotation marks to use it as a value. (line 1, character 11)]}]
        at com.atlassian.jira.rest.client.internal.async.AbstractAsynchronousRestClient$2.apply(AbstractAsynchronousRestClient.java:176)
        at com.atlassian.jira.rest.client.internal.async.AbstractAsynchronousRestClient$2.apply(AbstractAsynchronousRestClient.java:170)
        at com.atlassian.httpclient.api.ResponsePromiseMapFunction.apply(ResponsePromiseMapFunction.java:49)
        at com.atlassian.httpclient.api.ResponsePromiseMapFunction.apply(ResponsePromiseMapFunction.java:10)
        at io.atlassian.util.concurrent.Promises$OfStage.lambda$fold$4(Promises.java:332)
        at io.atlassian.util.concurrent.Promises.lambda$biFunction$7(Promises.java:422)
        at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930)
        at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
        at com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient$1.lambda$doCompleted$0(SettableFuturePromiseHttpPromiseAsyncClient.java:37)
        at com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient.runInContext(SettableFuturePromiseHttpPromiseAsyncClient.java:61)
        at com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient$ThreadLocalDelegateRunnable.run(SettableFuturePromiseHttpPromiseAsyncClient.java:129)
        ... 3 common frames omitted
```

Exception screenshots:

![image](https://user-images.githubusercontent.com/68207834/118668869-29b82c00-b7ed-11eb-8552-ad98a01726e7.png)

![image](https://user-images.githubusercontent.com/68207834/118669666-d692a900-b7ed-11eb-8322-414c9f1b7cef.png)

---

Also, escaping quotes or using encoded characters directly in the YAML (which shouldn't be needed) on the `jira.project`property config, like `%22` for quotes and `%4F%52` for the "OR" word, but it doesn't work, like in the config examples below, the exception is different this time too:  

```yaml
jira:
  - project:  \%22OR\%22
 (...)
```
![image](https://user-images.githubusercontent.com/68207834/118674563-c7adf580-b7f1-11eb-8b32-1f689ed290af.png)

```yaml
jira:
  - project: %4F%52
 (...)
````![image](https://user-images.githubusercontent.com/68207834/118673991-5e2de700-b7f1-11eb-90e8-70f01c2fae19.png)

```yaml
jira:
  - project: %22OR%22
(...)
```
![image](https://user-images.githubusercontent.com/68207834/118673837-3cccfb00-b7f1-11eb-9985-cb3e2f641d0b.png)
![image](https://user-images.githubusercontent.com/68207834/118674036-65ed8b80-b7f1-11eb-941f-7ac2cbbbe798.png)

### Environment Details

- CxFlow 1.9.20
- Windows 10
- CxFlow in either webhook mode or CLI
- JIRA Cloud

**Comment by cx-scord** [comment]
To try and help, the [JiraService.java](https://github.com/checkmarx-ltd/cx-flow/blob/master/src/main/java/com/checkmarx/flow/service/JiraService.java) is where the value is currently being set when creating the `GetCreateIssueMetadataOptions` instance with the project name and issue type with `withProjectKeys(...)` and `withIssueTypeNames(...)` [here](https://github.com/checkmarx-ltd/cx-flow/blob/master/src/main/java/com/checkmarx/flow/service/JiraService.java#L852), both the project and issue type are not being encoded before setting them.

**Comment by jbrotsos** [comment]
Released in 1.6.33
